zerver/lib/user_mutes.py

```diff
@@ -41,3 +42,15 @@ def remove_user_mute(user_profile: UserProfile, muted_user: UserProfile) -> None
 
 def user_is_muted(user_profile: UserProfile, muted_user: UserProfile) -> bool:
     return MutedUser.objects.filter(user_profile=user_profile, muted_user=muted_user).exists()
+
+
+@cache_with_key(get_muting_users_cache_key, timeout=3600 * 24 * 7)
+def get_muting_users(muted_user: UserProfile) -> Set[int]:
+    # This is kind of the inverse of `get_user_mutes` above.
+    # While `get_user_mutes` is mainly used for event system work,
+    # this is used in the message send codepath, to get a list
+    # of IDs of users who have muted a particular user.
```

Comment:
You can make this comment a docstring, and that'd probably be nicer.  Also, add another sentence mentioning that this will include deactivated users.  (I think that behavior is probably correct, given how we'll use this).

You can also move this commit to be a prep commit on the branch, since it should be mergeable independently.